docker-compose.yml

```
@@ -4,14 +4,50 @@ services:
     build:
       context: .
       target: ${DOCKER_TARGET:-development}
+      args:
+        - SKIP_BUILD_STATIC_GENERATION
+        - NEXT_PUBLIC_APP_ORIGIN
+        - NEXT_PUBLIC_API_BASE_URL
+        - NEXT_PUBLIC_UNIFIED_SEARCH_BASE_URL
+        - NEXT_PUBLIC_CMS_BASE_URL
+        - NEXT_PUBLIC_SENTRY_DSN
+        - NEXT_PUBLIC_ENVIRONMENT
+        - NEXT_PUBLIC_CAPTCHA_KEY
+        - NEXT_PUBLIC_LANGUAGE_CIMODE_VISIBLE
+        - NEXT_PUBLIC_HEADLESS_CMS_ENABLED
+        - NEXT_PUBLIC_FORMIK_PERSIST
+        - NEXT_PUBLIC_NEWSLETTER_ENABLED
+        - NEXT_PUBLIC_MATOMO_URL_BASE
+        - NEXT_PUBLIC_MATOMO_SITE_ID
+        - NEXT_PUBLIC_MATOMO_SRC_URL
+        - NEXT_PUBLIC_MATOMO_TRACKER_URL
+        - NEXT_PUBLIC_MATOMO_ENABLED
+        - NEXT_PUBLIC_CMS_TERMS_OF_SERVICE_SLUG_FI
+        - NEXT_PUBLIC_CMS_TERMS_OF_SERVICE_SLUG_EN
+        - NEXT_PUBLIC_CMS_TERMS_OF_SERVICE_SLUG_SV
+        - NEXT_PUBLIC_CMS_HEADER_MENU_NAME_FI
+        - NEXT_PUBLIC_CMS_HEADER_MENU_NAME_EN
+        - NEXT_PUBLIC_CMS_HEADER_MENU_NAME_SV
+        - NEXT_PUBLIC_CMS_FOOTER_MENU_NAME_FI
+        - NEXT_PUBLIC_CMS_FOOTER_MENU_NAME_EN
+        - NEXT_PUBLIC_CMS_FOOTER_MENU_NAME_SV
+        - PORT
+        - NEWSLETTER_BASE_URL
+        - NEWSLETTER_APIKEY
     environment:
       - WATCHPACK_POLLING=${WATCHPACK_POLLING:-true}
+      - PORT=${PORT:-3000}
+      - NEXT_PUBLIC_SENTRY_DSN=${NEXT_PUBLIC_SENTRY_DSN:-''}
+      - NEXT_PUBLIC_ENVIRONMENT=${DOCKER_TARGET:-development}
     volumes:
       # Share local directory to enable development with hot reloading
+      # NOTE: the volume mapping wil lbreak the production-stage build!
       - .:/app
       # Prevent sharing the following directories between host and container
       # to avoid ownership and/or platform issues:
-      - /app/node_modules
-      - /app/.next
+      # - /app/node_modules
+      # - /app/.next
```

```suggestion
      - /app/node_modules
      - /app/.next
```
These should not be removed as they are there for a reason: not to map the local file system's node_modules and .next to the Docker container. The installed node modules and .next should come from the Docker container, not from the local file system.